packages/advanced-logic/specs/payment-network-any-to-erc20-proxy-0.1.0.md

```diff
@@ -0,0 +1,314 @@
+
+This `paymentReference` consists of the last 8 bytes of a salted hash of the requestId: `last8Bytes(hash(lowercase(requestId + salt + address)))`:
+
+The contract also ensures that the `feeAmount` amount of the ERC20 transfer will be forwarded to the `feeAddress`.
+
+- `requestId` is the id of the request
+- `salt` is a random number with at least 8 bytes of randomness. It must be unique to each request
+- `address` is the payment address for payments, the refund address for refunds
+- `feeAmount` is the amount of the transfer that should be paid in fees
+- `feeAddress` is the address where the fee will be sent to
+- `conversion path` is a list of currency hashes to perform the conversion
+- `acceptedTokens` is the list of tokens addresses accepted for payments and refunds
+- `network` is the network of the tokens accepted for payments and refunds
+- `maxRateTimespan` is the time span maximum accepted between the payment and the rate timestamp
+- `lowercase()` transforms all characters to lowercase
+- `hash()` is a keccak256 hash function
+- `last8Bytes()` take the last 8 bytes
+
+As a payment network, this extension allows to deduce a payment `balance` for the request. (see
+[Interpretation](#Interpretation))
+
+## Contract
+
+The contract contains one function called `transferFromWithReferenceAndFee` which takes 6 arguments:
+
+- `to` is the destination address for the tokens
+- `requestAmount` is the amount to be paid in the request currency
+- `path` is the conversion path from the request currency to the payment token (see `conversion path`)
+- `paymentReference` is the reference data used to track the transfer (see `paymentReference`)
+- `feeAmount` is the amount of fees to be paid in the request currency
+- `feeAddress` is the destination address for the fee
+- `maxRateTimespan` is the time span maximum accepted between the payment and the rate timestamp
+
+The `TransferWithReferenceAndFee` event is emitted when the tokens are transfered. This event contains:
+
+- `tokenAddress` is the token address used for the payment
+- `to` is the destination address for the tokens
+- `requestCurrency` is the request currency hash
+- `paymentReference` is the reference data used to track the transfer (see `paymentReference`)
+- `feeAmount` is the amount of fees to be paid in the request currency
+- `feeAddress` is the destination address for the fee
+- `maxRateTimespan` is the time span maximum given for the payment
+
+[See smart contract source](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/smart-contracts/src/contracts/ProxyChainlinkConversionPath.sol)
+
+| Network | Contract Address                           |
+| ------- | ------------------------------------------ |
+| Mainnet | TODO                                       |
```

Comment:
Don't forget :) Maybe add a ticket in the backlog for when it's deployed?

Reply:
Good point